django_celery_fulldbresult/models.py

```diff
@@ -14,9 +25,49 @@
                             key=lambda t: t[0])
 
 
+def use_json():
+    """Returns True if django celery db result is configured to save results as
+    JSON.
+    """
+    return getattr(settings, "DJANGO_CELERY_FULLDBRESULT_USE_JSON", False)
+
+
+class PickledOrJSONObjectField(PickledObjectField):
+    """Serializes field content using pickle or JSON depending on the
+    DJANGO_CELERY_FULLDBRESULT_USE_JSON.
+    """
+
+    def get_db_prep_value(self, value, **kwargs):
+        if use_json():
+            if value is not None:
+                value = force_text(json.dumps(value))
+        else:
+            value = super(PickledOrJSONObjectField, self).get_db_prep_value(
+                value, **kwargs)
+
+        return value
+
+    def to_python(self, value):
+        if use_json():
+            if value is not None:
+                try:
+                    value = json.loads(value)
+                    return value
+                except Exception:
+                    if isinstance(value, str):
+                        # Badly formatted JSON!
```

Comment:
not really, this is a weird thing, but to_python is called with either a json representation OR a map or an array that is not yet serialized. if it's a string, it's safe to assume that it was intended to be serialized json that is badly serialized.